### ISSUE TITLE
GPU training does not progress with Tweedie loss

### ISSUE THREAD

**Issue by Rahkovsky** [catboost/catboost#2822]
GPU training on a small sample (642 records) fails to progress from iteration 0. I used a sample of severity insurance model data from [Kaggle AllState competition](https://www.kaggle.com/competitions/allstate-claims-severity). The model uses `CatBoostRegressor` with `Tweedie:variance_power=1.99` loss function. The same model works fine on CPU or if I use `RMSE` loss function instead of Tweedie. 

I use Python 3.11, Amazon Linux, `p3.2xlarge` AWS machine, and CatBoost 1.2.7.

```
INFO:__main__:Data shape (642, 116)
INFO:__main__:GPU estimation
INFO:__main__:Model fitted with 0 iterations
INFO:__main__:Validation score {'Tweedie:variance_power=1.99': 2817.0438144329896, 'RMSE': 3578.8665371485495}
INFO:__main__:CPU estimation
INFO:__main__:Model fitted with 605 iterations
INFO:__main__:Validation score {'Tweedie:variance_power=1.99': 109.1725731354047, 'RMSE': 3572.873741542586}
```

Replication code:

```
from typing import Tuple
import logging
import pandas as pd
from catboost import Pool
from catboost import CatBoostRegressor

logger = logging.getLogger(__name__)


def get_public_data() -> Tuple[pd.Series, Pool, list[str]]:
    """
    Get public ALlState Data for severity prediction
    https://www.kaggle.com/competitions/allstate-claims-severity/overview
    """
    df = pd.read_csv("severity_public_sample.csv")
    y_public = df["loss"]
    cat_columns = [c for c in df.columns if c.startswith("cat")]
    x_public = df[cat_columns]
    pool_public = Pool(data=x_public, label=y_public, cat_features=cat_columns)
    logger.info(f"Data shape {df[x_public.columns].shape}")
    return y_public, pool_public, cat_columns


def estimate_catboost(pool_train: Pool, task_type: str):
    power = 1.99
    model = CatBoostRegressor(cat_features=feature_names,
                              task_type=task_type,
                              loss_function=f'Tweedie:variance_power={power}',
                              use_best_model=True,
                              early_stopping_rounds=50,
                              custom_metric=['RMSE'],
                              eval_metric=f'Tweedie:variance_power={power}',
                              eval_fraction=0.15,
                              iterations=5000,
                              one_hot_max_size=2,
                              od_pval=0.001,
                              random_seed=42,  
                              logging_level='Silent',
                              )
    model.fit(pool_train)
    logger.info(f"Model fitted with {model.get_best_iteration()} iterations")
    logger.info(f"Validation score {model.get_best_score()['validation']}")


if __name__ == "__main__":
    y_public, pool_public, feature_names = get_public_data()
    logger.info(f"GPU estimation")
    estimate_catboost(pool_public, 'GPU')
    logger.info(f"CPU estimation")
    estimate_catboost(pool_public, 'CPU')
```

[severity_public_sample.csv](https://github.com/user-attachments/files/18985151/severity_public_sample.csv)

The same problem persists when I change the Tweedie power to 1.5 or 1.2. If we increase training sample size, the best iteration is still zero until about 30K records. Even with 30K records, the GPU model significantly under-trains relatively to CPU. The issue may be related to #2736.

**Comment by Rahkovsky** [comment]
I have tried to rerun this model with different versions of CatBoost. When I downgraded from version `1.2.7` to `1.2.5` the problem disappeared:

```(.env) [ec2-user@ip-172-18-105-43 boosting_behavior_models]$ python3.11 gpu_cpu_catboost_issue_public_data.py
INFO:__main__:Catboost version 1.2.5
INFO:__main__:Sample size 500
INFO:__main__:Data shape (500, 116)
INFO:__main__:GPU estimation
INFO:__main__:Model fitted with 2754 iterations
INFO:__main__:Validation score {'Tweedie:variance_power=1.99': 109.92424479166667, 'RMSE': 3954.951880027535}
INFO:__main__:CPU estimation
INFO:__main__:Model fitted with 756 iterations
INFO:__main__:Validation score {'Tweedie:variance_power=1.99': 109.32216269949586, 'RMSE': 3954.8285805826495}
INFO:__main__:Sample size 1000
INFO:__main__:Data shape (1000, 116)
INFO:__main__:GPU estimation
INFO:__main__:Model fitted with 349 iterations
INFO:__main__:Validation score {'Tweedie:variance_power=1.99': 109.72393229166667, 'RMSE': 4046.805410032297}
INFO:__main__:CPU estimation
INFO:__main__:Model fitted with 682 iterations
INFO:__main__:Validation score {'Tweedie:variance_power=1.99': 109.25403418402216, 'RMSE': 4047.073404102506}
```

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
@Rahkovsky 
on gpu catboost uses floats, and large labels cause numeric overflows, divergence etc. in tweedie regression
in this datatset, max label exceeds 10**5
if we normalize by max label, training on gpu and cpu produces models of similar accuracy

could you please check?

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
![Image](https://github.com/user-attachments/assets/78be7e8f-28f2-4fd8-8208-422955a1668e)

**Comment by Rahkovsky** [comment]
@Evgueni-Petrov-aka-espetrov, thank you for the explanation. I have confirmed that when I normalize data by the maximum label, I can run the CatBoost Tweeidie regression model on GPUs using CatBoost version 1.2.7.

At the same time, it would be great if CatBoost GPU training could work for large labels. The Tweeidie model is specifically designed to model the distribution of large extreme values. In version 1.2.5, GPU training did work with large values, so it appears this was possible before.
